src/main/java/org/apache/maven/plugins/clean/CleanMojo.java

```diff
@@ -161,6 +162,35 @@
     @Parameter( property = "maven.clean.excludeDefaultDirectories", defaultValue = "false" )
     private boolean excludeDefaultDirectories;
 
+    /**
+     * Enables fast clean if possible. If set to <code>true</code>, when the plugin is executed, a directory to
+     * be deleted will be atomically moved inside the <code>maven.clean.fastDir</code> directory and a thread will
+     * be launched to delete the needed files in the background.  When the build is completed, maven will wait
+     * until all the files have been deleted.  If any problem occurs during the atomic move of the directories,
+     * the plugin will default to the traditional deletion mechanism.
+     *
+     * @since 3.2
+     */
+    @Parameter( property = "maven.clean.fast", defaultValue = "false" )
+    private boolean fast;
+
+    /**
+     * When fast clean is specified, the <code>fastDir</code> property will be used as the location where directories
+     * to be deleted will be moved prior to background deletion.  If not specified, the
+     * <code>${maven.multiModuleProjectDirectory}/target/.clean</code> directory will be used.  If the
+     * <code>${build.directory}</code> has been modified, you'll have to adjust this property explicitly.
+     * In order for fast clean to work correctly, this directory must reside on the same mountpoint that the
+     * various directories that will be deleted.  This is usually the case when the whole code tree is setup to
+     * reside on a standard drive.
```

Comment:
I think the last sentence does not really makes sense. At most on Windows.

Reply:
Do you imply atomic move does not work on Windows ? I'm not sure what wording to use here... any suggestion welcomed !

Reply:
No, I don't but your decription reads weird. One sentence is talking about mountponits, but the subsequent one referring to the previous talkes about drives. It confuses me.

Reply:
Maybe you should use the term "volume"?

Reply:
s/drive/partition ?

Reply:
Use volume, this is platform-agnostic. Partition can be the same, but if you have a LVM or ZFS it will be another volume, datasets, etc.

Reply:
Do remote volume usually support atomic moves ?

Reply:
What about:
> In order for fast clean to work correctly, this directory and the various directories that will be deleted should usually reside on the same volume.  The exact conditions are system dependant though, but if an atomic move is not supported, the standard deletion mechanism will be used.

Reply:
I would expect within one volume, but not cross. Windows users don't even know that you cannot mount a partition into a directory on a different volume, that is why I want to keep the text OS/FS-agnostic.

Reply:
> What about:
> 
> > In order for fast clean to work correctly, this directory and the various directories that will be deleted should usually reside on the same volume.  The exact conditions are system dependant though, but if an atomic move is not supported, the standard deletion mechanism will be used.

Ideal.